pkg/filestore/s3/s3.go

```diff
@@ -0,0 +1,178 @@
+	if region == "" {
+		return nil, fmt.Errorf("region is required field")
+	}
+	if bucket == "" {
+		return nil, fmt.Errorf("bucket is required field")
+	}
+
+	s := &Store{
+		bucket: bucket,
+		logger: zap.NewNop(),
+	}
+	for _, opt := range opts {
+		opt(s)
+	}
+
+	sess, err := session.NewSession()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a session: %w", err)
+	}
+	creds := credentials.NewChainCredentials(
+		[]credentials.Provider{
+			&credentials.EnvProvider{},
+			&credentials.SharedCredentialsProvider{
+				Filename: credentialsFile,
+				Profile:  profile,
+			},
+			&ec2rolecreds.EC2RoleProvider{
+				Client: ec2metadata.New(sess),
+			},
+		},
+	)
+	cfg := aws.NewConfig().WithRegion(region).WithCredentials(creds)
+	s.client = s3.New(sess, cfg)
+
+	return s, nil
+}
+
+func (s *Store) NewReader(ctx context.Context, path string) (rc io.ReadCloser, err error) {
+	input := &s3.GetObjectInput{
+		Bucket: aws.String(s.bucket),
+		Key:    aws.String(path),
+	}
+	out, err := s.client.GetObjectWithContext(ctx, input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case s3.ErrCodeNoSuchKey:
+				err = filestore.ErrNotFound
+				return
+			case s3.ErrCodeInvalidObjectState:
+				err = fmt.Errorf("invalid object state: %w", err)
+				return
+			}
+		}
+		err = fmt.Errorf("unknown error given: %w", err)
+		return
+	}
+	return out.Body, nil
+}
+
+func (s *Store) GetObject(ctx context.Context, path string) (object filestore.Object, err error) {
+	object.Path = path
+	rc, err := s.NewReader(ctx, path)
+	if err != nil {
+		return
+	}
+	content, err := ioutil.ReadAll(rc)
+	if err != nil {
+		rc.Close()
+		return
+	}
+	err = rc.Close()
+	if err != nil {
+		return
+	}
+	object.Content = content
+	object.Size = int64(len(content))
+	return
+}
+
+func (s *Store) PutObject(ctx context.Context, path string, content []byte) error {
+	input := &s3.PutObjectInput{
+		Body:   aws.ReadSeekCloser(bytes.NewReader(content)),
+		Bucket: aws.String(s.bucket),
+		Key:    aws.String(path),
+	}
+	_, err := s.client.PutObjectWithContext(ctx, input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				return fmt.Errorf("error occured on aws side: %w", err)
+			}
+		}
+		return fmt.Errorf("unknown error given: %w", err)
+	}
+	return nil
+}
+
+func (s *Store) ListObjects(ctx context.Context, prefix string) ([]filestore.Object, error) {
+	var objects []filestore.Object
+	input := &s3.ListObjectsV2Input{
+		Bucket: aws.String(s.bucket),
+		Prefix: aws.String(prefix),
+	}
+
+	err := s.client.ListObjectsV2PagesWithContext(ctx, input,
+		func(page *s3.ListObjectsV2Output, lastPage bool) bool {
+			for _, obj := range page.Contents {
+				objects = append(objects, filestore.Object{
+					Path:    aws.StringValue(obj.Key),
+					Size:    aws.Int64Value(obj.Size),
+					Content: []byte{},
+				})
+			}
+			return *page.IsTruncated
+		})
```

Comment:
nit:

```
   },
)
```

Reply:
thank you, addressed on https://github.com/pipe-cd/pipe/pull/1485/commits/8dea680e82b2a373886f4c69bebc9c9fee066b03 🙏